File: ruby/lib/graphs.rb
class DirectedGraph
  class Node
    attr_accessor :value, :visited, :pointers

    def initialize(value, pointers = [])
      @value = value
      @visited = false
      @pointers = pointers
    end
  end

  attr_reader :nodes

  def initialize
    @nodes = []
  end

  def append(value, pointers = [])
    Node.new(value, pointers).tap do |node|
      @nodes << node
    end
  end
end

File: go/lib/graphs.go
package lib

type DGNode struct {
	Value    int
	Visited  bool
	Pointers []*DGNode
}

type DirectedGraph struct {
	Nodes []*DGNode
}

func NewDirectedGraph() *DirectedGraph {
	return &DirectedGraph{
		Nodes: make([]*DGNode, 0),
	}
}

func (g *DirectedGraph) Append(value int, pointers []*DGNode) *DGNode {
	node := &DGNode{
		Value:    value,
		Pointers: pointers,
	}
	g.Nodes = append(g.Nodes, node)
	return node
}
